libs/vkd3d/command.c

```diff
@@ -5362,8 +5362,22 @@ static void vk_image_copy_from_d3d12(VkImageCopy *image_copy,
     }
     else
     {
+        VkExtent3D dstExtent;
+
+        // Calculate the extent based on the source first
         unsigned int miplevel = image_copy->srcSubresource.mipLevel;
         vk_extent_3d_from_d3d12_miplevel(&image_copy->extent, src_desc, miplevel);
+
+        // Now make sure it does not extend past the destination when there is no src_box
+        miplevel = image_copy->dstSubresource.mipLevel;
```

Comment:
Just pass this directly to vk_extent_3d_from_d3d12_miplevel... Same for the above srcSubresource.